### PR TITLE
User dashboard trips

### PR DESCRIPTION
**Pull request by meyerhoferc**
Adds a listings table view for a host that links from their sidebar.
Adds a listing show view that links from the table
Adds a trips table index view that links from the user's sidebar
Adds a trip show view that links from a user's trips
A user cannot see another user's trip details

I'd like reviewers to especially look at the routes/controllers to see if we agree on how the nested controllers are setup.

RSpec: 76 examples, 1 failure, 1 pending

One test is currently failing, the test for a host editing a listing. This is failing because a the form for a listing is being done by @pixapi and hasn't been completed yet. I pushed up the test so that she would have it, and she told me she's going to work on making the Edit form for a listing tomorrow.

### CODE REVIEW

File: app/views/shared/_sidebar.html.erb [hunk 1]
```diff
@@ -25,7 +25,7 @@
                   </a>
               </li>
               <li>
-                  <a href="/users/<%= current_user.id %>/trips">
+                  <a href="<%= user_trips_path(current_user) %>">
```
Comment (janderson16) [inline app/views/shared/_sidebar.html.erb:28]: 👍

File: config/routes.rb [hunk 2]
```diff
@@ -15,9 +15,11 @@
   put "update_host", to: "users#update"
 
   resources :users, only: [:new, :create, :edit, :update] do
-    resources :messages
-    resources :trips
     resources :reservations
-    resources :listings
+    resources :messages
```
Comment (janderson16) [inline config/routes.rb:19]: this will be moved out from user whenever i push messages

Comment (meyerhoferc) [reply config/routes.rb:19]: good to know

File: config/routes.rb [hunk 3]
```diff
@@ -15,9 +15,11 @@
...
+    resources :trips, only: [:index, :show]
+    get 'listings', to: 'user/listings#index'
```
Comment (janderson16) [inline config/routes.rb:21]: 💯

**Comment by janderson16** [comment]
looks great, thank you for keeping it restful

**Pull request merged**